dms/models/directory.py

```diff
@@ -510,6 +516,26 @@ def _onchange_category(self):
             )
             record.tag_ids = tags
 
+    @api.onchange("model_id")
+    def _onchange_model_id(self):
+        for record in self:
+            if record.model_id:
+                record.res_model = record.model_id.model
```

Comment:
This is not something that should be done in an onchange (depends on the form), but in a compute (done always, at ORM level), right?. Related fields are computed fields behind the scenes. So, just make `res_model` a `fields.Char(related='model_id.model', readonly=True)`

Reply:
It's other important change that IMO it's necessary to apply but probably it's a better idea to apply in other PR but yes, will be related field, need to remove `_inverse_model_id` and simplify to `res_model` in `dms.file` (because it's other related about `directory_id.res_model`).
Now, i have added these function to prevent "inconsistent" data.

Reply:
As mentioned privately, this onchange can be removed because it overlaps with `_inverse_model_id()`:

https://github.com/OCA/dms/blob/d1ec8225c0a04de67fec79c2b2ed83dc75c37795/dms/models/directory.py#L315-L317

Reply:
No, it doesn't. This applies the change immediately, not on save. It's a usual technique for acting like a "compute".

Reply:
Anyway, the usual one is to call directly the inverse method, not to duplicate code.